core/src/main/scala/org/apache/spark/util/collection/Utils.scala

```diff
@@ -24,7 +26,8 @@ import com.google.common.collect.{Ordering => GuavaOrdering}
 /**
  * Utility functions for collections.
  */
-private[spark] object Utils {
+@Experimental
+object Utils {
```

Comment:
I think the other changes are borderline acceptable, but I am opposed to this one. This exposes methods like `tryOrStopSparkContext`, `setCustomHostname`, which could lead to undefined behavior. Most of the other methods here are highly specific for internal use where the semantics may not necessarily apply in a more general sense.